### PR TITLE
Refactor existing tests. Add integration tests for heap metrics

### PR DESCRIPTION
**Pull request by nisheedh**
*Fixes #, if available:*

*Description of changes:*
Refactor existing integration tests. Add integration tests for heap metrics

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    @Test
+    public void checkHeapInit() throws Exception {
+        checkHeapMetric(AllMetrics.HeapValue.HEAP_INIT, (d) -> d >= MIN_HEAP && d <= MAX_HEAP);
+    }
+
+    @Test
+    public void checkHeapMax() throws Exception {
+        checkHeapMetric(AllMetrics.HeapValue.HEAP_MAX, (d) -> d >= MIN_HEAP && d <= MAX_HEAP);
```
Comment (rguo-aws) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java:37]: is HeapMax supposed to be equivalent to MAX_HEAP ?

Comment (nisheedh) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java:37]: It'll depend on max memory allocated to elastic search JVM. In the current set up we have provided 512 MB for JVM.  I kept the check as a range, if we run on a environment with lower memory.

**Review by rguo-aws — APPROVED** [review]

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        Assert.assertEquals(1, responseData.getFieldDimensionSize());
+        Assert.assertEquals(metric.toString(), responseData.getField(0).getName());
+        Assert.assertEquals(JsonResponseField.Type.Constants.DOUBLE, responseData.getField(0).getType());
+        Assert.assertEquals(1, responseData.getRecordSize());
+        Double metricValue = responseData.getRecordAsDouble(0,metric.toString());
+        LOG.info("{}} value is {}", metric.toString(), metricValue);
+        Assert.assertTrue(metricValidator.test(metricValue));
+    }
+}
+
```
Comment (ktkrg) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java:77]: minor nit: extra line.

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.integ_test;
```
Comment (ktkrg) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java:1]: nit: Add copyright header.

**Review by ktkrg — APPROVED** [review]
The refactor and the tests look good. Just needs a copyright header on one of the files. If you're following it up with another PR, I'm ok approving this one.

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java (hunk 2 above)
Comment (nisheedh) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java:77]: Will do in a follow up CR

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java (hunk 3 above)
Comment (nisheedh) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/integ_test/HeapMetricsIT.java:1]: Will do in a follow up CR

**Pull request merged**